### ISSUE TITLE
samtools command error

### ISSUE THREAD

**Issue by JohnYu-yuzhongyi** [sanger-pathogens/circlator#66]
I'm trying to run a "circlator all assembly.fasta reads output_directory" and got an error a few minutes after the run begins:

> bam_sort] Use -T PREFIX / -o FILE to specify temporary and final output files
> Usage: samtools sort [options...] [in.bam]
> Options:
>   -l INT     Set compression level, from 0 (uncompressed) to 9 (best)
>   -m INT     Set maximum memory per thread; suffix K/M/G recognized [768M]
>   -n         Sort by read name
>   -o FILE    Write final output to FILE rather than standard output
>   -T PREFIX  Write temporary files to PREFIX.nnnn.bam
>   -@, --threads INT
>              Set number of sorting and compression threads [1]
>       --input-fmt-option OPT[=VAL]
>                Specify a single input file format option in the form
>                of OPTION or OPTION=VALUE
>   -O, --output-fmt FORMAT[,OPT[=VAL]]...
>                Specify output format (SAM, BAM, CRAM)
>       --output-fmt-option OPT[=VAL]
>                Specify a single output file format option in the form
>                of OPTION or OPTION=VALUE
>       --reference FILE
>                Reference sequence FASTA FILE [null]

I believe this error occurs because the command is missing a '-o'. I tried to run this command (with -o added) myself and it works. 

Thanks!

**Comment by martinghunt** [comment]
Could you please let me know the output of running

    circlator progcheck

Thanks.

**Comment by JohnYu-yuzhongyi** [comment]
> Using bwa version 0.7.13
> Using nucmer version 3.1
> Using prodigal version 2.6.2
> Using samtools version 1.3.1
> Using spades version 3.8.1

Thanks for your reply! I've fixed the problem later. I installed circlator using home brew so it automatically installed all the dependencies in their latest version. This I believe resulted in two errors. The first one is the problem I described above, that samtools needs a '-o'. What I did was I changed line 83 in ..../circlator/0.16.0_1/libexec/lib/python3.5/site-packages/circlator/mapping.py to:

> '-o', outfile

This helped me to get rid of the first problem. Then another problem occurred because my SPAdes version is too high. Although it seems does produce a fastg file but it's in a different file name. Luckily my home brew still kept the old version of SPAdes, so I "brew uninstall spades", and then "brew link spades", that gave me spades 3.6.0 back and circlator was able to run.

Hope these information helps! Thanks!
